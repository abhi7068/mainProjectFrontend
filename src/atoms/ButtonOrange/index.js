import React from 'react';

import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';

// re-defining material-ui classes for button
const useStyles = makeStyles((theme) => ({
  buttonOrange: {
    background: '#fb641b',
    boxShadow: '0 1px 2px 0 rgba(0,0,0,.2)',
    color: '#fff',
    fontWeight: '500',
    height: '48px',
  },
  contained: {
    '&:hover': {
      boxShadow: '0 4px 6px 0 rgba(0,0,0,.12)',
      background: '#fb641b',
      cursor: 'pointer',
    },
  },
}));

function ButtonOrange({ children, ...rest }) {
  const classes = useStyles();
  return (
    <Button
      data-testid="ButtonOrange"
      {...rest}
      className={classes.buttonOrange}
      classes={{ contained: classes.contained }}
    >
      {children}
    </Button>
  );
}
ButtonOrange.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
};
export default ButtonOrange;
