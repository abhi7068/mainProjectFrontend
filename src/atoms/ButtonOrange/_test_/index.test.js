import React from 'react';

import ButtonOrange from '../index';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

it('it renders', () => {
  render(<ButtonOrange />);
});

it('it renders with correct text', () => {
  const { getByTestId } = render(<ButtonOrange children="Submit" />);
  expect(getByTestId('button')).toHaveTextContent('Submit');
});

it('it matches snapshot', () => {
  const tree = renderer.create(<ButtonOrange children="Submit" />).toJSON();
  expect(tree).toMatchSnapshot();
});
