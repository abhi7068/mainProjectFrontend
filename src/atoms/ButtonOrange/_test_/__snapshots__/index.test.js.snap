// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttonOrange-89"
  data-testid="button"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    Submit
  </span>
</button>
`;
